test/parallel/test-http-socket-parser-null.js

```diff
@@ -1,7 +1,11 @@
 'use strict';
-// https://github.com/joyent/node/issues/4948
-
 require('../common');
+
+// This test makes sure node doesn't crash when a request is sent and the
+// connection is closed early due to `this.socket.parser` getting set to null
```

Comment:
I'm struggling to understand this description and test. I _think_ this is actually not testing anything as [`IncomingMessage.prototype._dump()`](https://github.com/nodejs/node/blob/599337f43e0e0d66263e69d70edab26b61d3038a/lib/_http_incoming.js#L294-L302) no longer touches `this.socket.parser`.

Reply:
You're right. It no longer touches the value if it's `null` already. I'll try to improve this comment further.